ImperatorToCK3/CommonUtils/ContainerField.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using commonItems;
+
+namespace ImperatorToCK3.CommonUtils {
+	public class ContainerField {
+		public ContainerField(List<string> initialValue) {
+			InitialValue = initialValue;
+		}
+		public List<string> GetValue(Date date) {
+			try {
```

Comment:
Exceptions seem like a not great way of handling flow control. I'd prefer to see explicit checks for the things that could go wrong. I'm guessing invalid date and missing value.

Reply:
I switched to checking if `pairsWithEarlierOrSameDate` is empty.